### PR TITLE
Update check_rmsThresh monitoring figure

### PR DESCRIPTION
**Pull request by cwnaber**
Example images using "main" defaults and "pretest" defaults, respectively.

image

image

### CODE REVIEW

File: experiment_helpers/check_rmsThresh.m [hunk 1]
```diff
@@ -1,30 +1,119 @@
...
+
+                % use middle 80%
+                onset = floor(onset + ((offset-onset)/10));
+                offset = ceil(offset - ((offset-onset)/10));
+                rmsValue = mean(data.rms(onset:offset, 1));
+
+                % can't determine rms mean, because no ost-based vowel found, and RMS too low
+            else
+                rmsValue = NaN;
+            end
```
Comment (carrien) [inline experiment_helpers/check_rmsThresh.m:89]: This is OK for today but I'd like to rework this so the rms calculation is not happening inside this `isgraphics` if statement. Only the plotting should happen in here. I'd suggest moving lines 63-89 before the isgraphics if statement, and replacing line 65 with something like:
```
[rmsValue,onset] = max(data.rms(:,1));
offset = offset;
```
or alternatively
```
rmsValue = max(data.rms(:,1));
onset = NaN;
offset = NaN;
```
Then you can end the switch/case. Afterwards, inside the isgraphics if statement, you can call the same plotting lines regardless of which case we're in.

File: experiment_helpers/check_rmsThresh.m [hunk 2]
```diff
@@ -1,30 +1,119 @@
...
+
+    %% set title and bGoodTrial
+    if rmsValue > params.rmsThresh
+        bGoodTrial = 1;
         title({'',''})
-    end %of GoodTrial conditional
+    else
+        bGoodTrial = 0;
+        title({'';'Amplitude below threshold!'})
+    end
```
Comment (carrien) [inline experiment_helpers/check_rmsThresh.m:115]: Similar to above; I don't think bGoodTrial should be set inside this isgraphics if statement. We should set it first, then plotting is separate. It doesn't change functionality so I'm okay making this adjustment later.

File: experiment_helpers/set_exptDefaults.m [hunk 3]
```diff
@@ -121,13 +121,24 @@
 durcalc.bMeasureOst = 0;
 expt = set_missingField(expt,'durcalc',durcalc);
 
-%% amplitude tracking parameters
+%% amplitude tracking parameters - old, mostly used with brut
```
Comment (carrien) [inline experiment_helpers/set_exptDefaults.m:124]: We should be using these in a lot of experiments with noise masking, so I would not add this comment.

File: experiment_helpers/set_exptDefaults.m [hunk 4]
```diff
@@ -121,13 +121,24 @@
...
+amp.checkMethod = 'mean';   % Compare mean RMS during vowel against amp.rmsThresh
+amp.rmsThresh = 0.037;      % RMS values below this trigger "speak louder" prompt. 0.037 is ~78.5 dBA on SMNG hardware
+    % expt.amp.limits is a 2x2 array structured like this:
+    %        [GoodLow, GoodHi;
+    %         WarnLow, WarnHi]
+    % In check_rmsThresh, a line is drawn in green between the low and hi Good limits,
+    % and a line in yellow between the Warn limits.
+amp.limits = [0.037, 0.100; 0 0];
+expt = set_missingField(expt, 'amp', amp);
+
```
Comment (carrien) [inline experiment_helpers/set_exptDefaults.m:141]: Can we just add these fields to amplcalc instead of creating "amp"?

**Review by carrien — CHANGES_REQUESTED** [review]

**Comment by cwnaber** [comment]
@carrien More revisions to address your comments. Individual commits for each situation, though the diff'er had some trouble with the reorganizing commit. Let me know if this is what you had in mind.

**Comment by cwnaber** [comment]
In `check_rmsThresh`, using `params.checkMode = 'peak'` now plots a magenta circle on the axis at the peak.

image

**Review by carrien — APPROVED** [review]

**Pull request merged**